data-serving/data-service/src/controllers/preprocessor.ts

```diff
@@ -0,0 +1,82 @@
+import { Case, CaseDocument } from '../model/case';
+import { NextFunction, Request, Response } from 'express';
+
+import { RevisionMetadata } from '../model/revision-metadata';
+
+const createNewMetadata = (curatorEmail: string, creationNotes: string) => {
+    return {
+        revisionNumber: 0,
+        creationMetadata: {
+            curator: curatorEmail,
+            date: Date.now(),
+            notes: creationNotes,
+        },
+    };
+};
+
+const createUpdateMetadata = (
+    c: CaseDocument,
+    curatorEmail: string,
+    updateNotes: string,
+) => {
+    return {
+        revisionNumber: ++c.revisionMetadata.revisionNumber,
```

Comment:
For some reason I had in my mind that this was a property of `updateMetadata` (which itself was a repeated field). But that level of audit detail probably isn't necessary.

Reply:
`updateMetadata` isn't repeated -- might be strange, but the `revisionMetadata` field isn't meant to capture the full edit log, only the first edit (i.e. `creationMetadata`) and most recent (`updateMetadata`), so that we can display these easily in the UI. (You can imagine wanting to see who initially curated the case and when it was most recently updated & why.)

The full log in theory would be available through the versioning pattern which has not yet been implemented :)

Does that make sense?

Reply:
Totally; mostly just musing. That was approximately my understanding.

Reply:
👍